### ISSUE TITLE
[gatsby-plugin-mdx][gatsby-remark-images] generates new img-url on build

### ISSUE THREAD

**Issue by mschel** [gatsbyjs/gatsby#21622]
## Relevant information

When adding images in MDX Files using relative links (every content has it's own folder including the `.mdx` file and all assets) every time I run `gatsby build` the URL to the image is slightly different. For example:

`![Labor](labor.jpg)`

Build 1: `http://localhost/static/7259c46d503debb82dbd9fe1e50bdcf4/e7068/labor.jpg`
Build 2: `http://localhost/static/7259c46d503debb82dbd9fe1e50bdcf4/eea4a/labor.jpg`

The last URL segment changes all the time. This makes it hard to share the content (we are using this for news) on Facebook, as all images are gone in old posts after deploying a new build.

All other assets are rendered with the same URL all the time .

### Plugin Config
```
{
      resolve: `gatsby-plugin-mdx`,
      options: {
        defaultLayouts: {
          default: require.resolve("./src/components/layout.js"),
        },
        gatsbyRemarkPlugins: [
          {
            resolve: "gatsby-remark-images",
            options: {
              maxWidth: 1920,
              linkImagesToOriginal: false,
            },
          },
        ],
        plugins: [{ resolve: "gatsby-remark-images" }],
      },
    },
```
### Environment 
Everything is on `latest`
- gatsby-remark-images@^3.1.44
- gatsby-plugin-mdx@^1.0.71
- gatsby-core-utils@^1.0.28
- gatsby-cli@^2.8.29:

Thanks for helping! Cheers!

**Comment by freiksenet** [comment]
Hello!

I tried it with `examples/using-remark` and images seem to be stable.

**Comment by mschel** [comment]
Hi @freiksenet - thank you for getting into this. Is the example based on the remark plugin in combination with mdx rendering? The problem only occurs in this combination - all other images rendered with remark are fine and have stable urls.

**Comment by mschel** [comment]
It seems like the URL change appeared after an update of the `gatsby-remark-images` plugin (`3.1.39` to `3.1.44`). We will try to get a minimal reproduction running and validate this guess.

**Comment by freiksenet** [comment]
I'm not sure what could have affected this :( I wonder if you could share the source code of your repo with us? If you can't do it publicly, you can email me to freiksenet@gatsbyjs.com.

**Comment by mschel** [comment]
We tried to reproduce the behaviour by setting the `remark-images` version to `3.1.39` - well this did not change the URL at all. Afterwards we restored the `yarn.lock` file before the latest `yarn upgrade`. Now the URL gets a different hash value. So it seems like it is not `remark-images` on it's own - but a combination of several updated plugins. I've attached both versions (old and new) of the lock file.

[yarn-lock.zip](https://github.com/gatsbyjs/gatsby/files/4244795/yarn-lock.zip)
